File: src/Enterprise/C#/patterns/Iterator.cs
using System.Collections.Generic;
using System.Linq;
namespace Genkidama.PatternExamples;
public static class IteratorExample { public static bool Run(){using var it=new List<int>{10,20,30}.GetEnumerator();var seen=new List<int>();while(it.MoveNext())seen.Add(it.Current);return seen.SequenceEqual(new[]{10,20,30})&&!it.MoveNext();} }

File: src/Enterprise/VB.NET/patterns/Iterator.vb
Imports System.Collections.Generic
Imports System.Linq
Friend Module IteratorExample
    Friend Function Run() As Boolean
        Dim values={10,20,30},seen As New List(Of Integer)
        For Each value In values : seen.Add(value) : Next
        Return seen.SequenceEqual(values)
    End Function
End Module
